cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -369,11 +369,15 @@ func StartControllers(s *options.CMServer, kubeconfig *restclient.Config, rootCl
 	// Find the list of namespaced resources via discovery that the namespace controller must manage
 	namespaceKubeClient := client("namespace-controller")
 	namespaceClientPool := dynamic.NewClientPool(restclient.AddUserAgent(kubeconfig, "namespace-controller"), restMapper, dynamic.LegacyAPIPathResolverFunc)
```

Comment:
I don't think so?  We recreate a new dynamic client for each `GroupVersionResource`

https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/namespace/namespace_controller_utils.go#L282

Which I think should be capable of handling it w/o a new mapper?  But I have to admit this part of the codebase is pretty opaque...